### DISCUSSION TITLE
unable to update field value using setValue() or resetField() in mui

### DISCUSSION THREAD

**Discussion by deepakchandrawanshi1063** [react-hook-form/react-hook-form#8761] [Q&A]
import React, { useRef } from "react";
import {
  TextField,
  Button,
  Slide,
  Box,
  FormControlLabel,
  FormControl,
  FormLabel,
  Radio,
  RadioGroup,
  Grid
} from "@material-ui/core";
import { useForm, Controller } from "react-hook-form";

const SuperDelux = () => {
  const initialValue = {
    Wmachine: "",
    WmType: "",
    washroom: "",
    area: "",
    cloths: ""
  };
  const {
    register,
    watch,
    reset,
    setValue,
    resetField,
    formState: { errors },
    handleSubmit,
    control
  } = useForm({ defaultValues: initialValue });

  const onClickSubmit = (data) => {
    console.log("formdata", data);
  };
  const [Wmachine, WmType, washroom, area, cloths] = watch([
    "Wmachine",
    "WmType",
    "washroom",
    "area",
    "cloths"
  ]);

  console.log("watchWashingMachine", Wmachine, WmType);
  const no = document.getElementById("no");
  if (no) {
    if (no.checked) {
      setValue(WmType, "");
      //resetField(WmType)
    }
  }

  return (
    <>
      <p>SuperDelux Form</p>
      <form onSubmit={handleSubmit(onClickSubmit)}>
        <Grid container>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              placeholder="Enter number of washroom"
              {...register("washroom")}
            />
          </Grid>
          <Grid item xs={12}>
            {washroom && (
              <TextField
                variant="outlined"
                placeholder="Enter area in sqft."
                {...register("area")}
              />
            )}
          </Grid>
          <Grid item xs={12}>
            {area && (
              <TextField
                variant="outlined"
                placeholder="Enter no. of cloths"
                {...register("cloths")}
              />
            )}
          </Grid>
          <Grid item xs={12}>
            <br />
            {cloths && (
              <Controller
                control={control}
                name="Wmachine"
                render={({ field: { onChange, onBlur, value, ref } }) => (
                  <FormControl>
                    <FormLabel id="demo-radio-buttons-group-label">
                      Washing Machile available
                    </FormLabel>
                    <RadioGroup
                      aria-labelledby="demo-radio-buttons-group-label"
                      name="Wmachine"
                      onChange={onChange}
                      onBlur={onBlur}
                      selected={value}
                      row
                    >
                      <FormControlLabel
                        value="yes"
                        control={<Radio />}
                        label="Yes"
                      />
                      <FormControlLabel
                        value="no"
                        control={<Radio id="no" />}
                        label="No"
                      />
                    </RadioGroup>
                  </FormControl>
                )}
              />
            )}
          </Grid>
          <Grid item xs={12}>
            <br />
            {Wmachine === "yes" && (
              <Controller
                control={control}
                name="WmType"
                render={({ field: { onChange, onBlur, value, ref } }) => (
                  <FormControl>
                    <FormLabel id="demo-radio-buttons-group-label">
                      Washing Machile type
                    </FormLabel>
                    <RadioGroup
                      aria-labelledby="demo-radio-buttons-group-label"
                      name="WmType"
                      row
                      onChange={onChange}
                      onBlur={onBlur}
                      selected={value}
                    >
                      <FormControlLabel
                        value="manual"
                        control={<Radio />}
                        label="Manual"
                      />
                      <FormControlLabel
                        value="automatic"
                        control={<Radio />}
                        label="Automatic"
                      />
                    </RadioGroup>
                  </FormControl>
                )}
              />
            )}
          </Grid>
          <Grid item xs={12}>
            <Button variant="outlined" onClick={() => setValue(WmType, "")}>
              Click
            </Button>
          </Grid>
        </Grid>

        <Button type="submit" variant="contained" color="primary">
          submit
        </Button>
      </form>
    </>
  );
};

export default SuperDelux;

Result
=========
formdata :{
Wmachine: "no"
WmType: "manual"
washroom: "1"
area: "152"
cloths: "34"
manual: ""}

=> when I am selecting Wmachine to 'yes' and 'WmType'  to "automatic" or "manual" type. then when I click on Wmachine to 'no' the WmType value is not going to change. I want to set WmType to ""(empty) when i am click on no.

Codesandbox link:
===================
[https://codesandbox.io/s/conditional-form-p8dkwr?file=/src/component/SuperDelux.js:0-5632](url)

**Comment by Moshyfawn** [comment]
Please, take your time to explain what's your desired behavior and the steps to reproduce the issue you're having.
